core/datatrak.c

```diff
@@ -32,12 +32,12 @@ static unsigned long four_bytes_to_long(unsigned char x, unsigned char y, unsign
 	return ((long)x << 24) + ((long)y << 16) + ((long)z << 8) + (long)t;
 }
 
-static unsigned char *byte_to_bits(unsigned char byte)
+static unsigned char *byte_to_bits(unsigned char byte_value)
```

Comment:
Hmm.. I generally find these fixes of not too much value, because I don't see how such a short function would be confusing.

In this case though, wouldn't it make *much* more sense to simply remove the global `byte` variable?

[Rant: It's really vexing that you can't return `char [8]` in C or C++.]

Reply:
I agree that your approach is much better - remove the global.

The value of static checkers has been hotly debated for 20+ years. I finally found one that can be reasonably integrated with github and it seemed a relatively small price to pay to get a bit more robustness in place.

BTW: there are a couple Alerts in its findings that I would love for you to take a look.
https://lgtm.com/projects/g/Subsurface-divelog/subsurface/alerts/?mode=list
e.g. "Redefined default parameter", "Resource not released in destructor", "Virtual call from constructor to destructor"
I am happy to ignore some of the warnings (like the ones about FIXME comments), but I think it's worth while to at least look at the others and understand if they represent an actual concern or not.
Thanks :-)

Reply:
I am currently looking at some of the alerts, not the ones you mentioned, though...

Reply:
to avoid duplicating work, feel free to point out what you are looking at :-)